### PR TITLE
Prince Leopold Island (combined)

### LINKED ISSUE #818: Prince Leopold Island
**Topic:** [Prince Leopold Island](https://github.com/juliadenham/Summit_knowledge/blob/main/prince_leopold_island.md) is an island in the Qikiqtaaluk Region of Nunavut, Canada. It is located in Lancaster Sound at the junction of Prince Regent Inlet and Barrow Strait.

**Folder Location:** taxonomy/knowledge/geography/islands/prince_island

**Downloads:** 
- qna.yaml: [prince_leopold_island.txt](https://github.com/instructlab/taxonomy/files/15226922/prince_leopold_island.txt)
- [attribution.txt](https://github.com/instructlab/taxonomy/files/15226298/prince-leopold-island-attribution.txt) 

**Wiki link:** https://en.wikipedia.org/wiki/Prince_Leopold_Island

### PR DESCRIPTION
**Pull request by ckadner**
**Describe the contribution to the taxonomy**

Combined knowledge submissions for Prince Leopold Island from Red Hat Summit 2024.

Closes #902
Closes #972
Closes #981
Closes #996

Resolves #818

FYI @amaa1981, @fatehks, @webdba, @youeyoue

**Contribution checklist**

- [ ] The contribution was tested with `ilab generate`
- [ ] No errors or warnings were produced by `ilab generate`
- [x] All [commits are signed off](https://github.com/instructlab/taxonomy/blob/main/CONTRIBUTING.md#legal) (DCO)
- [x] The `qna.yaml` file contains at least 5 `seed_examples`
- [x] The `qna.yaml` file was [linted](https://yamllint.com) and [prettified](https://onlineyamltools.com/prettify-yaml) ([yaml-validator](https://jsonformatter.org/yaml-validator) can do both)
- [x] An `attribution.txt` file in the same folder as the `qna.yaml` file
- [x] Content does not include PII or otherwise sensitive or confidential information
- [x] Content does not include anything documented in the project's [Avoid these Topics](https://github.com/instructlab/taxonomy/blob/main/docs/SKILLS_GUIDE.md#avoid-these-topics) guidelines

### CODE REVIEW

**Comment by ckadner** [comment]
@kelbrown20 @juliadenham -- seems like the Markdown version of the wikipedia page is missing some numbers like the height of the cliffs surrounding the island

Wikipedia:

> ## Geography and Geology
> 
> Most of Prince Leopold Island rises steeply from the surrounding sea,
> with cliffs rising as **high as 265 m (869 ft) above sea** level surrounding
> most of the island.


Extracted Mardown:


> ## Geography and Geology
> 
> Most of Prince Leopold Island rises steeply from the surrounding sea,
> with cliffs rising as **high as above sea level** surrounding most of the
> island.

**Comment by ckadner** [comment]
@obuzek -- I made an effort to straighten out the qna pairing and make the questions as precise as possible.

However, some of the question-answer pairs are semantically correct, but aren't an exact match to the text found in the Markdown (or Wikipedia). Is that problematic?

File: knowledge/geography/islands/prince_island/qna.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    most of the island.
+  question: How high are the cliffs surrounding Prince Leopold Island?
+- answer: The island and Somerset Island to its south were first sighted by a
+    European on August 4, 1819, during William Parry's 1819 expedition.
+  question: Prince Leopold Island was first sighted during which expedition?
+- answer: Large numbers of arctic seabirds, thick-billed murres, northern fulmars,
+    black-legged kittiwakes, and black guillemots breed on the cliff ledges, arriving
+    in the vicinity in May or early June and departing by mid-September.
+  question: What animals live on Prince Leopold Island?
+- answer: "74°N 90°W"
```
Comment (ckadner) [inline knowledge/geography/islands/prince_island/qna.yaml:24]: This information is missing from the Markdown version

https://github.com/instructlab/taxonomy/pull/1154#issuecomment-2146437322

File: knowledge/geography/islands/prince_island/qna.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  question: What are the coordinates of Prince Leopold Island?
+- answer: It is located in Lancaster Sound at the junction of Prince Regent
+    Inlet and Barrow Strait.
+  question: What body of water surrounds Prince Leopold Island?
+- answer: Prince Leopold Island is a significant breeding ground for large
+    populations of several arctic bird species.
+  question: For what species of animals is Prince Leopold Island a significant
+    summer habitat for?
+- answer: The majority of the island's area of 68 km2 (26 sq mi) is occupied by a
```
Comment (ckadner) [inline knowledge/geography/islands/prince_island/qna.yaml:33]: This information is missing from the Markdown version

https://github.com/instructlab/taxonomy/pull/1154#issuecomment-2146437322

File: knowledge/geography/islands/prince_island/qna.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    plateau generally ranging in altitude between around 240 m (790 ft) and 300 m
+    (980 ft).
+  question: What is the altitude of the plateau that makes up most of Prince
+    Leopold Island?
+- answer: Vegetation consists of forbs, lichens, and mosses, though the top of the
+    island is mostly devoid of plants. Denser concentrations of vegetation occur in
+    seepage areas on the spits, including sedges and some dwarf willow
+  question: What does the flora consist of on Prince Leopold Island?
+- answer: The majority of the island's area of 68 km2 (26 sq mi) is occupied by a
```
Comment (ckadner) [inline knowledge/geography/islands/prince_island/qna.yaml:42]: This information is missing from the Markdown version

https://github.com/instructlab/taxonomy/pull/1154#issuecomment-2146437322

File: knowledge/geography/islands/prince_island/qna.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    north, southwest, south, and southeast of the island, jut out from the coast.
+    During the summer, surface water on the island is scarce. A few small shallow
+    lakes exist on the plateau; the largest, Far Lake, is about 0.5 km (0.31 mi)
+    long, and is located near the eastern cliffs
+  question: What is the terrain like on Prince Leopold Island?
+- answer: Large numbers of arctic seabirds, thick-billed murres, northern fulmars,
+    black-legged kittiwakes, and black guillemots breed on the cliff ledges, arriving
+    in the vicinity in May or early June and departing by mid-September.
+  question: What types of birds live on Prince Leopold Island?
+- answer: 306 m (1004 ft)
```
Comment (ckadner) [inline knowledge/geography/islands/prince_island/qna.yaml:60]: This information is missing from the Markdown version

https://github.com/instructlab/taxonomy/pull/1154#issuecomment-2146437322

**Comment by ckadner** [comment]
https://github.com/instructlab/taxonomy/issues/1160

File: knowledge/geography/islands/prince_island/qna.yaml (hunk 4 above)
Comment (ckadner) [reply knowledge/geography/islands/prince_island/qna.yaml:60]: @juliadenham updated the Markdown in commit 706f365a256950f408707e2f03fe76e380c734c6

**Comment by ckadner** [comment]
@instructlab-bot precheck

**Comment by ckadner** [comment]
@instructlab-bot generate

**Comment by ckadner** [comment]
Previous set of generated questions and answers seemed very different from the those in the qna.yaml

Giving it another go with a simplified `task_description`

**Comment by ckadner** [comment]
@instructlab-bot generate

**Comment by ckadner** [comment]
FYI @bjhargrave -- I replaced the `task_description` with a much simpler version and it does not impact the generated questions at all 🤷🏻

**Comment by ckadner** [comment]
Trying SDG once more with unrelated `task_description`

**Comment by ckadner** [comment]
@instructlab-bot generate

**Comment by ckadner** [comment]
@instructlab-bot generate

**Comment by ckadner** [comment]
slightly differently worded but semantically equivalent questions, none of them duck related 🦆

**Comment by ckadner** [comment]
@instructlab-bot generate

**Comment by ckadner** [comment]
Restoring original `qna.yaml`

**Comment by ckadner** [comment]
@instructlab-bot generate

**Comment by bjhargrave** [comment]
@ckadner Did some changes finally improve the SDG?

**Comment by jjasghar** [comment]
Hi! 👋
It’s been a while since you’ve seen any movement on this PR. We haven’t forgotten about you!  We’ve run into some logistical issues, hence this delay. We absolutely want your PR, and being marketed as `e2e-ready` is still the last stop before we get it into the upstream model.

We are thankful for your patience and ask that you please keep this PR open. As soon as we finish all our behind-the-scenes work, we’ll test the full model against your submissions and, ideally, accept your amazing contribution(s)! 

Your Community Maintainer Team.

P.S. if you have any specific questions or thoughts, don’t hesitate to comment on pull request this or email [awesome@instructlab.ai](mailto:awesome@instructlab.ai) and [lhawthor@redhat.com](mailto:lhawthor@redhat.com), and we’ll get back to you as soon as possible.

**Comment by jjasghar** [comment]
Hi! :wave:
With the new update and release of 0.18.0, our [base knowledge submissions](https://github.com/instructlab/taxonomy/?tab=readme-ov-file#knowledge-yaml-examples) have drastically changed. This has given us an opportunity to significantly improve the accuracy of our fine-tuned models, which is a major win for our community!

But alas, that has caused some changes that this PR isn't able to continue forward. We greatly appreciate the work you've put into this, and wish we could accept it but the amount of work to get it to be `version: 3` is unrealistic for us to ask.
If you _do_ want to do this, please don't hesitate to take the `qna.yaml` update it, and build it out. Otherwise, we hope you come back to our community with a new submission and help build out our Open Source community-driven LLM.

If you have any questions directly, please don't hesitate to comment on the PR or email me at [awesomeATinstructlab.ai](http://awesomeatinstructlab.ai/) (yes, really), and I'll do my best to answer your questions.

**Pull request closed without merging**